### PR TITLE
Fixed issue to allow running reggie.py from any folder.

### PR DESCRIPTION
**Pull request by earthwise01**
Fixed module_path not returning the correct path if run from outside `Reggie-Updated` folder.

### TL;DR
`module_path` only returned the correct path when running `reggie.py` from inside the `Reggie-Updated` folder. This made `reggie.py` act as if the `reggiedata` folder was missing when running from outside the `Reggie-Updated` folder. 
**Fixed using `pathlib`.**

### Bug Details
This bug was present on both MacOS 10.14.6 & Ubuntu 20.04, Based on the code the issue should also have occurred on Windows. This fix has been tested on both MacOS 10.14.6 & Ubuntu 20.04, and has worked on both operating systems.

The following shows the original behaviour of `module_path`. The expected output of '2' should have been `/Users/joshua/PycharmProjects/Reggie-Updated/`. reggie.py was run from `/Users/joshua/PycharmProjects/` and the `Reggie-Updated` folder was located under `PycharmProjects`.

Here's an example of how I diagnosed the problem.
**The debugging code has since been removed.**
```
File: reggie.py
Function: def module_path():

    if __name__ == '__main__':
        print("1====================================")
        print(sys.argv[0])
        print("2====================================")
        print(os.path.abspath(sys.argv[0]))
        return os.path.dirname(os.path.abspath(sys.argv[0]))
```
Here is the output of `reggie.py` when ran from `/Users/joshua/PycharmProjects/` with the above debugging code included. This is wrong because the folder in which `reggie.py` is located is named `Reggie-Updated` and not `Reggie! Level Editor`. `module_path` was supposed to return the absolute path of `reggie.py` but instead returns the current working directory path + `Reggie! Level Editor`. Because of this, the program only works if launched from the `Reggie-Updated` folder.
```
1====================================
Reggie! Level Editor
2====================================
/Users/joshua/PycharmProjects/Reggie! Level Editor
```
This an example of the error reggie gives when run outside of the `Reggie-Updated` folder:
Screen Shot 2020-09-13 at 2 48 28 PM

These problems have been resolved in the pull request.

### To reproduce(on Linux/MacOS):

1. Git clone Reggie-Updated & `git reset --hard 7f9a67a`
2. cd to an directory outside the folder. Eg. `cd Reggie-Updated/..` 
3. Run reggie.py from outside the folder. Eg.`python3 Reggie-Updated/reggie.py` This should produce the above error.
4. Compare this with running it *inside* the folder. Eg. `cd Reggie-Updated; python3 ./reggie.py` This should work as expected.

### CODE REVIEW

**Comment by RoadrunnerWMC** [comment]
Thanks for the very detailed PR!

Unfortunately, though, this fix implementation isn't going to work. It works great on Python 3, but I'm trying to keep this code compatible with Python 2 as well, and pathlib was only introduced in 3.4.

The reason `sys.argv[0]` doesn't point to the correct location is the following line, which I added relatively recently:

```python
sys.argv[0] = ApplicationDisplayName  # only way to set the app display name on Qt 4
```

This is what ultimately broke the current-folder-detection code. `__file__`, however, should still point to reggie.py.

I love pathlib a lot and use it in most of my new projects, but as I explained, I don't want to use it in Reggie-Updated. Thus, I propose the following fix, which I've successfully tested with both Python 2 and 3:

```python
return os.path.dirname(os.path.abspath(__file__))
```

Does that sound good to you?

**Comment by earthwise01** [comment]
Yes, I just tested your fix on MacOS with Python 3.8.2 and it works as expected. I’d say I’d be a good idea to implement the fix(`return os.path.dirname(os.path.abspath(__file__))`).

**Comment by RoadrunnerWMC** [comment]
There we go, then! Thanks again!
